### PR TITLE
Some additions..

### PR DESCRIPTION
**Pull request by derekcollison**
Added in async error callback with defaults eprintln!
Made Message is_no_responders() public for normal use.

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: src/options.rs [hunk 1]
```diff
@@ -667,3 +693,26 @@ impl ReconnectDelayCallback {
         self.0(reconnects)
     }
 }
+
+pub(crate) struct ErrCallback(Option<Box<dyn Fn(Error) + Send + Sync + 'static>>);
```
Comment (caspervonb) [inline src/options.rs:697]: Nit (`Err` and `std::Error` are different things)

```suggestion
pub(crate) struct ErrorCallback(Option<Box<dyn Fn(Error) + Send + Sync + 'static>>);
```

File: src/options.rs [hunk 2]
```diff
@@ -667,3 +693,26 @@ impl ReconnectDelayCallback {
...
+impl ErrCallback {
+    pub fn call(&self, client: &Client, err: Error) {
+        if let Some(callback) = self.0.as_ref() {
+            callback(err);
+        } else {
+            let si = client.server_info();
+            eprintln!("{} on connection [{}]", err, si.client_id);
```
Comment (caspervonb) [inline src/options.rs:704]: Maybe we should call into `log`(which is configurable) instead of printing directly to stderr here.

**Review by caspervonb — COMMENTED** [review]
Minor nits

File: src/options.rs (hunk 2 above)
Comment (derekcollison) [reply src/options.rs:704]: This is a default version, we could call into log though if we already have it in the client..

Comment (derekcollison) [reply src/options.rs:704]: So with log.error! default does not print anything and I want the default handler, meaning if you do not set one, to print to stderr when bad things happen.

**Review by caspervonb — APPROVED** [review]

**Pull request merged**